modules/layers/src/solid-polygon-layer/polygon.ts

```diff
@@ -305,15 +335,55 @@ export function getSurfaceIndices(
   if (preproject) {
     // When tesselating lnglat coordinates, project them to the common space for accuracy
     const n = positions.length;
+
     // Clone the array
     positions = positions.slice();
+
     const p: number[] = [];
     for (let i = 0; i < n; i += positionSize) {
       p[0] = positions[i];
       p[1] = positions[i + 1];
+
+      if (full3d) {
```

Comment:
`positionSize === 3` should be required here too?

Reply:
PositionSize is calculated in the tesselator, but the preproject function is passed in from the solid-polygon-layer which doesnt have access to this so always sets the preproject to the xyz variant when full3d is true. So in here when full3d is true we are always going to have the xyz variant of the preproject function.
Ofc if the projectPosition class works with xy and xyz data, then we need to add the positionSize===3 here too, but I assumed it didn't as the argument didn't seem to suggest it.

Reply:
Ignore me, I just looked and I see it supports both. Shall change.